rpcs3/rpcs3qt/settings_dialog.cpp

```diff
@@ -782,6 +782,82 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> guiSettings, std:
 		EnableBufferingOptions(enabled && ui->enableBuffering->isChecked());
 	};
 
+	auto ChangeMicrophoneType = [=](QString text)
+	{
+		std::string s_standard, s_singstar, s_realsingstar, s_rocksmith;
+
+		auto enableMicsCombo = [=](u32 max)
+		{
+			ui->microphone1Box->setEnabled(true);
+
+			if (max == 1 || ui->microphone1Box->currentText() == xemu_settings->m_microphone_creator.mic_none)
+				return;
+
+			ui->microphone2Box->setEnabled(true);
+
+			if (max > 2 && ui->microphone2Box->currentText() != xemu_settings->m_microphone_creator.mic_none)
+			{
+				ui->microphone3Box->setEnabled(true);
+				if (ui->microphone3Box->currentText() != xemu_settings->m_microphone_creator.mic_none)
+				{
+					ui->microphone4Box->setEnabled(true);
+				}
+			}
+		};
+
+		ui->microphone1Box->setEnabled(false);
+		ui->microphone2Box->setEnabled(false);
+		ui->microphone3Box->setEnabled(false);
+		ui->microphone4Box->setEnabled(false);
+
+		fmt_class_string<microphone_handler>::format(s_standard, static_cast<u64>(microphone_handler::standard));
+		fmt_class_string<microphone_handler>::format(s_singstar, static_cast<u64>(microphone_handler::singstar));
+		fmt_class_string<microphone_handler>::format(s_realsingstar, static_cast<u64>(microphone_handler::real_singstar));
+		fmt_class_string<microphone_handler>::format(s_rocksmith, static_cast<u64>(microphone_handler::rocksmith));
+
+		if (text == s_standard.c_str())
+		{
+			enableMicsCombo(4);
+			return;
+		}
+		if (text == s_singstar.c_str())
+		{
+			enableMicsCombo(2);
+			return;
+		}
+		if (text == s_realsingstar.c_str() || text == s_rocksmith.c_str())
+		{
+			enableMicsCombo(1);
+			return;
+		}
+	};
+	auto PropagateUsedDevices = [=]()
```

Comment:
Space these lambdas a little. I missed this being a separate function the first time.